scripts/zephyr_module.py

```diff
@@ -95,10 +95,43 @@
     type: seq
     sequence:
       - type: str
+  blobs:
+    required: false
+    type: seq
+    sequence:
+      - type: map
+        mapping:
+          path:
+            required: true
+            type: str
+          sha256:
+            required: true
+            type: str
+          type:
+            required: true
+            type: str
+            enum: ['img', 'lib']
```

Comment:
we have two types, `img` and `lib`, but I don't see how those types are used or why they are there.

Do we need to differentiate and why ?
I assume that somewhere in the build system itself (CMake) some dedicated code for handling a particular binary blob will be written, and that specific code will know what to do with a given blob, regardless of the type specified in the yaml blob.

Or do you envision some generic handler that can automatically perform an action if the bin blob is of a dedicated type in the future.

Reply:
Right now this is exclusively for the users' information. Every time a user types `west list` they will see the type(s) of blob each HAL contains, and then it's up to them to decide whether they want to use blobs or not.

Reply:
thanks for confirming my observation.

I don't have a problem with the field, as it purely provides informational value, but curious if it then should be required or optional.
Of course using required from start, avoids everyone later having to update their blob yaml.
Allowing us to easier implement generic use of the field as all blobs already contains the field.
Having `enum` does provide uniform way of describing this field (avoiding differences like `lib` vs `library`).
On the other hand, for something being informational, then we need to update the `enum` every time it's not precise enough.

Consider someone adding both `32-bit` and `64-bit` version, they might like to have: `lib32` and `lib64` as available types.

So trying to find the right balance of not over-engineering from start for something we are not using today (and thus might need to adjust anyway when we start to use it), while at the same time being sufficient prepared for the future.

Reply:
I think `lib` and `img` are a good starting point. I doubt very much that we will need to differentiate between 32 and 64-bit in the actual type, but this can instead be achieved by adding a new `arch` field in the schema in the future, which I think makes more sense than using the `type` field.

Reply:
> I doubt very much that we will need to differentiate between 32 and 64-bit in the actual type, but this can instead be achieved by adding a new arch field in the schema in the future

unfortunately my :crystal_ball: is broken.